session.go

```diff
@@ -30,27 +30,49 @@ func (s *SessionClient) GetSession(
 		return nil, errors.New("PoktNodeSessionFetcher not set")
 	}
 
-	req := &sessiontypes.QueryGetSessionRequest{
-		ApplicationAddress: appAddress,
-		ServiceId:          serviceId,
-		BlockHeight:        height,
+	// Enforce any deadlines specified on the supplied context.
+	// Create channel for result
+	type result struct {
+		session *sessiontypes.Session
+		err     error
 	}
 
-	// TODO_IMPROVE: Would it be feasible to add a GetCurrentSession, supported by the underlying protocol?
-	//
-	// It seems likely that GetSession will almost always be used to get the session
-	// matching the latest height.
-	//
-	// In addition, the current session that is being returned could:
-	// - Include the latest block height
-	// - Reduce the number of SDK calls needed for sending relays
-	// - Remove the need for the BlockClient
-	res, err := s.PoktNodeSessionFetcher.GetSession(ctx, req)
-	if err != nil {
-		return nil, err
-	}
+	resultCh := make(chan result, 1)
+	// Launch GetApplication in goroutine
+	go func() {
+		req := &sessiontypes.QueryGetSessionRequest{
+			ApplicationAddress: appAddress,
+			ServiceId:          serviceId,
+			BlockHeight:        height,
+		}
 
-	return res.Session, nil
+		// TODO_TECHDEBT(@adshmh): consider increasing the default response size:
+		// e.g. using google.golang.org/grpc's MaxCallRecvMsgSize CallOption.
+		//
+		// TODO_IMPROVE: Would it be feasible to add a GetCurrentSession, supported by the underlying protocol?
```

Comment:
https://github.com/pokt-network/poktroll/blob/6afdff0b0d41c49677e62edad2af3de5c2506cc2/x/session/keeper/query_get_session.go#L16

<img width="883" alt="Screenshot 2025-04-16 at 1 59 43 PM" src="https://github.com/user-attachments/assets/91f13f85-8da9-4379-a17b-8eee12d33ddc" />

1. Can you try this?
2. If it works -> remove the TODO
3. If it doesn't work -> Please open an issue in ght poktroll repo because it's a bug

Reply:
Thank you for the review. This TODO was just moved around, not added.
It is now removed. This is already how PATH uses the SDK:

https://github.com/buildwithgrove/path/blob/4d94e2e6bfea5a4774c3a7263c0cce69338d95b4/protocol/shannon/fullnode_lazy.go#L94-L100

![image](https://github.com/user-attachments/assets/3ecf17e5-67ba-4c1b-8b43-3972aa89f4e6)